cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/LocalApplicationTemplateManager.java

```diff
@@ -112,7 +112,7 @@ public LocalApplicationTemplateManager(CConfiguration configuration, PipelineFac
                                                     queueAdmin, discoveryServiceClient));
     pipeline.addLast(new ProgramGenerationStage(configuration, namespacedLocationFactory));
     pipeline.addLast(new ApplicationRegistrationStage(store));
-    pipeline.setFinally(new EnableConcurrentRunsStage(preferencesStore));
+    pipeline.addLast(new EnableConcurrentRunsStage(preferencesStore));
```

Comment:
So we don't want the last stage to be executed if there is an error thrown during the any of the preceding stages. Is that the reasoning behind this change?